vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc

```diff
@@ -455,4 +456,37 @@ void Wallet::OnTransferAnonToExternalWalletAddress(
       transfer_callback);
 }
 
+void Wallet::GetGrantViaSafetynetCheck(const std::string& promotion_id) {
+  ledger_->LoadURL(braveledger_bat_helper::buildURL(
+      (std::string)GET_PROMOTION_ATTESTATION + ledger_->GetPaymentId(),
+      PREFIX_V1),
+  std::vector<std::string>(), "", "",
+  ledger::URL_METHOD::GET, std::bind(&Wallet::GetGrantViaSafetynetCheckCallback,
+                                   this,
+                                   promotion_id,
+                                   _1,
+                                   _2,
+                                   _3));
+}
+
+void Wallet::GetGrantViaSafetynetCheckCallback(const std::string& promotion_id,
+                                                  int response_status_code,
+                                                  const std::string& response,
+                                                  const std::map<std::string,
+                                                  std::string>& headers) {
+  ledger_->LogResponse(__func__, response_status_code, response, headers);
+
+  if (response_status_code != net::HTTP_OK) {
+    // Attestation failed
```

Comment:
Do we need this comment?

Reply:
yes

Reply:
Resolving as code for lines 481 to 483 can be moved to a function such as `AttestationFailed` for potential re-use and testing